aptos-move/framework/aptos-framework/sources/primary_fungible_store.move

```diff
@@ -61,10 +64,22 @@ module aptos_framework::primary_fungible_store {
         owner: address,
         metadata: Object<T>,
     ): Object<FungibleStore> acquires DeriveRefPod {
-        if (!primary_store_exists(owner, metadata)) {
+        let store_addr = primary_store_address(owner, metadata);
+        if (fungible_asset::store_exists(store_addr)) {
+            object::address_to_object<FungibleStore>(store_addr)
+        } else {
             create_primary_store(owner, metadata)
+        }
+    }
+
+    inline fun apt_ensure_primary_store_exists(
```

Comment:
Why is this inline? A long inline function used multiple times = bloating up the bytecode

Reply:
I made not huge private functions inline, to make things faster/cheaper, and was suggested to inline more things on other PRs.
What's the threshold for "function being too long for inline" in your view? here it is like if + 4 function calls, and below one is if + 2 function calls

Reply:
> bloating up the bytecode

@movekevin why is this a problem? The Aptos Framework does not face the same package size constraints as dApps, so what is the concern with trading off more bytecode side for less execution gas?

Reply:
> bloat the bytecode

@movekevin it it appropriate to consider increased bytecode size "bloat" if the bytecode size is a function of reducing gas size?

What is the problem with people looking at the framework for an example of gas-optimized code, especially for critical execution paths?

You can always add in a func comment explaining that things are inlined for the sake of performance

Reply:
https://github.com/aptos-labs/aptos-core/pull/13194#issuecomment-2099438733

and in general - inline is always done for performance, there is no other reasons to inline.